python/mujinplanningclient/binpickingplanningclient.py

```diff
@@ -11,68 +11,97 @@
+            goals (str): (deprecated) flat list of goals, e.g. two 5d ik goals: [380,450,50,0,0,1, 380,450,50,0,0,-1]
 
             targetnamepattern (str): regular expression describing the name of the object. No default will be provided, caller must set this. See https://docs.python.org/2/library/re.html
             approachoffset (float, optional): Distance in millimeters to move straight to the grasp point, e.g. 30 mm
             departoffsetdir (list[float], optional): The direction and distance in mm to move the part in global frame (usually along negative gravity) after it is grasped, e.g. [0,0,50]
             destdepartoffsetdir (list[float], optional): The direction and distance in mm to move away from the object after it is placed, e.g. [0,0,30]. Depending on leaveoffsetintool parameter, this can in the global coordinate system or tool coordinate system.
-            leaveoffsetintool (int, optional): If 1, destdepartoffsetdir is in the tool coordinate system. If 0, destdepartoffsetdir is in the global coordinate system. By default this is 0.
             deletetarget (int, optional): whether to delete target after pick and place is done
-            toolname (str, optional): Name of the manipulator
-            regionname (str, optional): Name of the region of the objects
-            cameranames (list[str], optional): The names of the cameras to avoid occlusions with the robot
-            envclearance (float, optional): Environment clearance in millimeters
-
-        Low level planning parameters:
             debuglevel (str): sets debug level of the task
             movetodestination (str): planning parameter
             freeinc (str): planning parameter
             worksteplength (str): planning parameter
             densowavearmgroup (str): planning parameter
+            regionname (str, optional): Name of the region of the objects
+            cameranames (list[str], optional): The names of the cameras to avoid occlusions with the robot
+            envclearance (float, optional): Environment clearance in millimeters
+            toolname (str, optional): Name of the manipulator
+            robotspeed (float, optional): Value in (0,1] defining the percentage of speed the robot should move at.
+            timeout (float, optional): Time in seconds after which the command is assumed to have failed. (Default: 1000)
+            leaveoffsetintool (int, optional): If 1, destdepartoffsetdir is in the tool coordinate system. If 0, destdepartoffsetdir is in the global coordinate system. By default this is 0.
+            desttargetname (str): The destination target name where the destination goal ikparams come from
+            destikparamnames (str): A list of lists of ikparam names for the destinations of the target. Only destikparamnames[0] is looked at and tells the system to place the part in any of the ikparams in destikparamnames[0]
```

Comment:
This type doesn't look right

Reply:
Oh, I agree. Fixed the spec. 

Tho its the same on master. For those big docsting functions like PickAndPlace I didnt make any actual changes to the docstring in this MR. Only reordered them. Reason is if you reorder and change stuff - the diff becomes really hard to read. So I decided to reorder them first and then make a separate followup MR.

I still think it is a good idea, so lets leave this one as is for now.